cluster-up/cluster/kind/common.sh

```diff
@@ -304,6 +304,6 @@ function down() {
         return
     fi
     $KIND delete cluster --name=${CLUSTER_NAME}
-    docker rm -f $REGISTRY_NAME >> /dev/null
+    docker rm -f $REGISTRY_NAME >> /dev/null || true
```

Comment:
I remember vaguely, that overriding a value in a function will eliminate the `-e` in it totally,
so if for example the previous commands fail they won't fail the test as well

maybe we are find with it in this case
if we are not, we can check if this is the behavior (for example add `false` line above it)
and consider trying something like  `! docker rm -f $REGISTRY_NAME >> /dev/null`
that will achieve same effect but might still respect the `-e`

Reply:
Thanks for the review @oshoval !
- I don't think your concern about eliminating `-e` is a problem here, as the `||` only applies to the one command and not an entire function
- `! <cmd>` would indeed succeed on failure, but also fail on success, so that's a problem :)
- Completely agree on `>` vs `>>`, but it works so I was going to leave it alone
- I think originally you mentioned that I was modifying the wrong line, and you're absolutely right, it must be the `kind` command above that calls `docker rm`, I should probably do both

Reply:
Thanks

> Thanks for the review @oshoval !
> 
> * I don't think your concern about eliminating `-e` is a problem here, as the `||` only applies to the one command and not an entire function

We saw at the past, that masking one line affect the whole function, even that it isn't logical in the first place, it is a gotcha of bash

> * `! <cmd>` would indeed succeed on failure, but also fail on success, so that's a problem :)

It won't fail on success, nothing checks the return code, because you add `!` only, 
it is syntactic sugar for `|| true`

Reply:
> Thanks
> 
> > Thanks for the review @oshoval !
> > 
> > * I don't think your concern about eliminating `-e` is a problem here, as the `||` only applies to the one command and not an entire function
> 
> We saw at the past, that masking one line affect the whole function, even that it isn't logical in the first place, it is a gotcha of bash (not sure it is exactly the case here, but generally speaking)
> 
> > * `! <cmd>` would indeed succeed on failure, but also fail on success, so that's a problem :)
> 
> It won't fail on success, nothing checks the return code, because you add `!` only, it is syntactic sugar for `|| true` (but anyhow it is needed only if the above is relevant here)

Indeed, my bad! I had to write a little test to convince myself:
``` 
$ cat test.sh 
#!/bin/bash -e

test() {
	echo 1
	false || true
	echo 2
	! true
	echo 3
	! false
	echo 4
	false
	echo 5
}

test
$ ./test.sh 
1
2
3
4
$ 
```